### PR TITLE
Upgrade logback-classic to address CVE CVE-2023-6378

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Force resolve logback-classic to 1.3.12 to address https://github.com/advisories/GHSA-vmq6-5m68-f53m

### Issues Resolved
[List any issues this PR will resolve]

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by cwperks — APPROVED** [review]

**Review by DarshitChanpura — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -497,6 +497,7 @@ configurations {
             force "org.apache.httpcomponents:httpcore:4.4.16"
             force "com.google.errorprone:error_prone_annotations:2.23.0"
             force "org.checkerframework:checker-qual:3.40.0"
+            force "ch.qos.logback:logback-classic:1.3.12"
```
Comment (reta) [inline build.gradle:500]: @cwperks this should probably be set to `1.2.13` since Logback 1.3.x needs SLF4J 2.0 APIs

Comment (cwperks) [reply build.gradle:500]: Thank you reta! How can we catch issues like this as part of CI?

From logback's [README](https://github.com/qos-ch/logback#java-ee-and-jakarta-ee-versions):

> Both 1.3.x and 1.4.x series require SLF4J 2.0.x or later.

Comment (cwperks) [reply build.gradle:500]: Is the issue preventing slf4j upgrade that OpenSearch still supports Java 8? From the slf4j [README](https://github.com/qos-ch/slf4j#how-to-build-slf4j)

> SLF4J version 2.0.x will run under Java 8 but requires Java 9 or later to build.

Comment (reta) [reply build.gradle:500]: > Is the issue preventing slf4j upgrade that OpenSearch still supports Java 8?

We basically could move to slf4j 2.x but I believe some deps (like Jetty 9 fe, if I am not mistaken)  do still rely on slf4j 1.x

**Pull request merged**